alameda.js

```diff
@@ -111,10 +111,10 @@ var requirejs, require, define;
         shim: {},
         config: {}
       },
-      mapCache = {},
+      mapCache = Object.create(null),
       requireDeferreds = [],
-      deferreds = {},
-      calledDefine = {},
+      deferreds = Object.create(null),
+      calledDefine = Object.create(null),
       calledPlugin = {},
```

Comment:
It seems like we should sweep all of these object literal properties for consistency.

In general I am on the fence on this change: if it helps in performance, I suppose it would be fine, although maybe make a utility method, maybe `emptyObj()` that does the `Object.create(null)` work. You said hasProp was showing up, with this change, how did it help those traces?